CMakeLists.txt

```diff
@@ -21,9 +21,11 @@ else()
   message(STATUS "Building in ${CMAKE_BUILD_TYPE} configuration")
 endif()
 
-set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
 set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} -O3")
 set(CMAKE_C_FLAGS_RELEASE   "${CMAKE_C_FLAGS_RELEASE}   -O3")
+if(APPLE)
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -std=c++11")
```

Comment:
It doesn't build on my side on linux without this flag. So in fact, it seems to be required on all platform.
It seems that `CMAKE_CXX_FLAGS` is propagated to CUDA NVCC FLAGS in a way.
And the `set(CMAKE_CXX_STANDARD 11)` has no impact on the NVCC FLAGS.

Reply:
There are also lots of target_* calls in src/CMakeLists.txt that should be deprecated but are required for CUDA. Simone found out that all of that was needed.